aptos-move/framework/aptos-stdlib/sources/type_info.move

```diff
@@ -23,6 +25,32 @@ module aptos_std::type_info {
 
     public native fun type_name<T>(): string::String;
 
+    /// Return the size, in bytes, of a `T`, provided an immutable
+    /// reference to a quasi-null instance of fixed-size `T`,
+    /// `fixed_size_type_null_ref`.
+    ///
+    /// Analogous to `sizeof()` in C.
+    ///
+    /// Ideally this function would be implemented as a native function
+    /// of the form `public native fun size_of<T>(): u64;`, such that
+    /// callers do not need to concoct quasi-null instances, e.g.
+    /// `false` for `T` as a `bool`, `0` for `T` as a `u8`, or `@0x0`
+    /// for `T` as an `address`.
+    ///
+    /// Does not actually enfore that `T` is a fixed-size type, which
+    /// would require determining whether or not vectors are nested
+    /// within.
+    ///
+    /// See `test_size_of()` for an analysis of common types and nesting
+    /// patterns, as well as `test_size_of_vectors()` for an analysis of
+    /// vector size dynamism.
+    public fun size_of<T>(
+        fixed_size_type_null_ref: &T
```

Comment:
nit: This can fit nicely on a single line

Reply:
This function is problematic in multiple ways.

- I do not understand the documentation. What is `fixed_size_type_null_ref` mean? What does `null` mean in the context of Move? I do not think users have a chance to understand, because these things aren't defined in Move.
- The function says in its own documentation that it should not exists like it exists but be a native function (which then likely has a different semantics). So why don't we do it right? Once this is a public function in aptos stdlib we can never change it.
- If this is for accessing size in memory, the function is unfeasible. For example, a u128 with value `0` maybe just a byte serialized but much larger in Memory. A struct can be serialized a few bytes but expands to 100+ bytes in memory. 

Lets please do this right.